battle-service/src/main/java/com/company/battle/controllers/GameController.java

```diff
@@ -0,0 +1,68 @@
+package com.company.battle.controllers;
+
+import com.company.common.dtos.GameDto;
+import com.company.common.dtos.SettingsDto;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("api/v1/game")
+public class GameController {
+
+    @GetMapping("/{id}")
+    GameDto getGameById(@PathVariable UUID id) {
+        return null;
+    }
+
+    @GetMapping("/{link}")
+    GameDto getGameByLink(@PathVariable String link) {
+        return null;
+    }
+
+    @GetMapping
+    List<GameDto> getActiveGames() {
+        return null;
+    }
+
+    @PostMapping("/create")
+    GameDto createGame(@RequestBody GameDto gameDto) {
+        return null;
+    }
+
+    @PutMapping("/join")
+    GameDto joinGame(@RequestBody String link) {
+        return null;
+    }
+
+    @PutMapping("/leave")
+    GameDto leaveGame(@RequestBody GameDto gameDto) {
+        return null;
+    }
+
+    @GetMapping("/{id}")
+    SettingsDto getSettingsByGameId(@PathVariable UUID id) {
+        return null;
+    }
+
+    @PostMapping("/default")
+    SettingsDto addDefaultSettings() {
+        return null;
+    }
+
+    @PostMapping("/custom")
+    SettingsDto addCustomSettings(@RequestBody SettingsDto settingsDto) {
```

Comment:
The same as previous, if settings is part of game domain. You can use post patch put for the whole game entity